src/azul/plugins/metadata/hca/transform.py

```diff
@@ -81,6 +84,42 @@
 pass_thru_uuid4: PassThrough[api.UUID4] = PassThrough()
 
 
+class OrganismAge(FieldType[JSON, str]):
+
+    def to_index(self, value: JSON) -> str:
+        try:
+            value, unit = value['value'], value['unit']
+        except TypeError:  # Used for translating None
+            value, unit = [None, None]
+        except KeyError:
+            require(value.keys() == {'value', 'unit'},
+                    'A dictionary with `value` and `unit` keys is required')
+            value, unit = [None, None]
+        else:
+            reject(value == '',
+                   'The key `value` must not use an empty string value')
+            reject(unit == '',
+                   'The key `unit` must not use an empty string value')
+        if value is None:
+            require(unit is None,
+                    'A `unit` key without `value` is not valid')
+            return NullableString.null_string
+        elif unit is None:
+            return value
+        else:
+            return ' '.join(i.strip() for i in [value, unit])
```

Comment:
> I think we should reject age range values/units with leading or trailing spaces instead of silently removing them.

Understood, I was attempting to follow [your directive](https://github.com/DataBiosphere/azul/pull/1909#discussion_r467214177).

> Is there existing metadata in production that has those?

No, I did not find any.

Reply:
Granted, my directive

> Since we use space as the joiner we need to ensure it doesn't occur in value or unit.

could have been interpreted either way, actively removing of spaces or rejecting them.

But as you know, we prefer strictness over leniency and in that context there is no ambiguity. Actively removing spaces is lenient and therefore to be avoided.